### PR TITLE
Feature/add typescript and prettier support

### PR DESCRIPTION
**Pull request by mwoodcertinia**
* Update the rules to support Typescript and Prettier
* Remove all warnings: either error or turn the rule off

### CODE REVIEW

**Review by dfruddffdc — APPROVED** [review]

**Review by pwrightcertinia — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # @financialforcedev/eslint-config
 
+## 4.0.0
```
Comment (dbtavernerffdc) [inline CHANGELOG.md:3]: Sorry I don't get why version 4.0.0 is in ChangeLog 3?

File: index.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-/**
- * Copyright (c) 2017, FinancialForce.com, inc
+/*
+ * Copyright (c) 2017-2020, FinancialForce.com, inc
```
Comment (dbtavernerffdc) [inline index.js:2]: For future reference the rules on copyright notices have changed and so there is no need to do this anymore. Just the very first date on its own should do.

File: CHANGELOG.md (hunk 1 above)
Comment (pwrightcertinia) [reply CHANGELOG.md:3]: It's been renamed to CHANGELOG.md, we now just put everything into one file 👍

File: index/src/javascript/error.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2020, FinancialForce.com, inc
```
Comment (dbtavernerffdc) [inline index/src/javascript/error.js:2]: I wouldn't go back and change any of them but I find these copyrights a little verbose and they mark the code slightly harder to read. I would go for the short version where we can.

Comment (dfruddffdc) [reply index/src/javascript/error.js:2]: We should clarify with Legal, but I believe the copyright policy is different for open source repositories. I think we're supposed to include the full copyright notice with correct dates,

**Review by dbtavernerffdc — APPROVED** [review]

File: index/src/javascript/error.js (hunk 3 above)
Comment (mwoodcertinia) [reply index/src/javascript/error.js:2]: Open source has to be this notice: (https://github.com/financialforcedev/eslint-config/blob/master/LICENSE)

**Pull request merged**